client/tests/test_updating_provider.py

```diff
@@ -0,0 +1,153 @@
+import featureform as ff
+from featureform.resources import ResourceRedefinedError
+import json
+import pytest
+from types import SimpleNamespace
+
+
+@pytest.mark.parametrize(
+    "provider_source_fxt,is_local,is_insecure",
+    [
+        pytest.param("local_provider_source", True, True, marks=pytest.mark.local),
+        pytest.param("hosted_sql_provider_and_source", False, False, marks=pytest.mark.hosted),
+        pytest.param("hosted_sql_provider_and_source", False, True, marks=pytest.mark.docker),
+    ]
+)
+def test_valid_provider_update(provider_source_fxt, is_local, is_insecure, request):
+    custom_marks = [mark.name for mark in request.node.own_markers if mark.name != 'parametrize']
+    provider, source, inference_store = request.getfixturevalue(provider_source_fxt)(custom_marks)
+
+    # Arranges the resources context following the Quickstart pattern
+    resource_client, postgres_name, redis_name = arrange_resources(provider, source, inference_store, is_local, is_insecure)
+
+    # DESCRIPTION
+    postgres_description = "This is an updated description for Provider A"
+    redis_description = "This is an updated description for Provider B"
+    # USERNAME
+    postgres_username = "username_a_updated"
+    # PASSWORD
+    postgres_password = "password_a_updated"
+    redis_password = "password_b_updated"
+
+    if is_local:
+        postgres_host = "0.0.0.0"
+        redis_host = "0.0.0.0"
+    else:
+        # The host name for postgres is different between Docker and Minikube
+        postgres_host = "host.docker.internal" if "docker" in custom_marks else "quickstart-postgres"
+        redis_host = "host.docker.internal" if "docker" in custom_marks else "quickstart-redis"
+
+    postgres = ff.register_postgres(
+        name=postgres_name,
+        host=postgres_host,
+        port="5432",
+        database="postgres",
+        user=postgres_username,
+        password=postgres_password,
+        description=postgres_description
+    )
+
+    redis = ff.register_redis(
+        name=redis_name,
+        host=redis_host,
+        port=6379,
+        password=redis_password,
+        description=redis_description
+    )
+
+    resource_client.apply()
```

Comment:
Feel like this will likely always error in localmode right?

Reply:
Due to the postgres/redis providers or another reason?

Reply:
nm was thinking about serving